pkg/azuredisk/azure.go

```diff
@@ -20,26 +20,52 @@ import (
 	"fmt"
 	"os"
 
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 	"k8s.io/klog"
 	"k8s.io/legacy-cloud-providers/azure"
 )
 
 // GetCloudProvider get Azure Cloud Provider
 func GetCloudProvider() (*azure.Cloud, error) {
-	credFile, ok := os.LookupEnv("AZURE_CREDENTIAL_FILE")
-	if ok {
-		klog.V(2).Infof("AZURE_CREDENTIAL_FILE env var set as %v", credFile)
-	} else {
-		credFile = "/etc/kubernetes/azure.json"
-		klog.V(2).Infof("use default AZURE_CREDENTIAL_FILE env var: %v", credFile)
+	klog.V(2).Infof("reading cloud config from secret")
+	kubeClient, err := getKubeClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get KubeClient: %v", err)
+	}
+	az := &azure.Cloud{
+		KubeClient: kubeClient,
 	}
+	az.InitializeCloudFromSecret()
 
-	f, err := os.Open(credFile)
-	if err != nil {
-		klog.Errorf("Failed to load config from file: %s", credFile)
-		return nil, fmt.Errorf("Failed to load config from file: %s, cloud not get azure cloud provider", credFile)
+	if az.ResourceGroup == "" {
```

Comment:
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/docs/cloud-provider-config.md#auth-configs
use `az.SubscriptionId` to tell whether it's already InitializeCloudFromSecret

Reply:
done